java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/impl/DefaultDeploymentTemplateReplacements.java

```diff
@@ -101,26 +109,36 @@ protected Map<String, String> getAppDefinitionData(AppDefinition appDefinition)
     }
 
     protected Map<String, String> getEnvironmentVariables(AppDefinition appDefinition, Session session) {
-	Map<String, String> environmentVariables = getEnvironmentVariables(Optional.of(session));
+	Map<String, String> environmentVariables = getEnvironmentVariables(appDefinition, Optional.of(session));
 	environmentVariables.put(PLACEHOLDER_ENV_SESSION_URL,
 		TheiaCloudDeploymentUtil.getSessionURL(ingressPathProvider, appDefinition, session));
 	return environmentVariables;
     }
 
     protected Map<String, String> getEnvironmentVariables(AppDefinition appDefinition, int instance) {
-	Map<String, String> environmentVariables = getEnvironmentVariables(Optional.empty());
+	Map<String, String> environmentVariables = getEnvironmentVariables(appDefinition, Optional.empty());
 	environmentVariables.put(PLACEHOLDER_ENV_SESSION_URL,
 		TheiaCloudDeploymentUtil.getSessionURL(ingressPathProvider, appDefinition, instance));
 	return environmentVariables;
     }
 
-    protected Map<String, String> getEnvironmentVariables(Optional<Session> session) {
+    protected Map<String, String> getEnvironmentVariables(AppDefinition appDefinition, Optional<Session> session) {
 	Map<String, String> environmentVariables = new LinkedHashMap<String, String>();
 	environmentVariables.put(PLACEHOLDER_ENV_APP_ID, arguments.getAppId());
 	environmentVariables.put(PLACEHOLDER_ENV_SERVICE_URL, arguments.getServiceUrl());
 	environmentVariables.put(PLACEHOLDER_ENV_SESSION_UID, session.map(s -> s.getMetadata().getUid()).orElse(""));
 	environmentVariables.put(PLACEHOLDER_ENV_SESSION_NAME, session.map(s -> s.getSpec().getName()).orElse(""));
 	environmentVariables.put(PLACEHOLDER_ENV_SESSION_USER, session.map(s -> s.getSpec().getUser()).orElse(""));
+	environmentVariables.put(PLACEHOLDER_ENV_SESSION_SECRET,
+		session.map(s -> s.getSpec().getSessionSecret()).orElse(""));
+	if (arguments.isEnableMonitor()) {
+	    if (appDefinition.getSpec().getMonitor() != null && appDefinition.getSpec().getMonitor().getPort() != 0) {
```

Comment:
Just a minor improvement to the check because negative ports do not make sense ;)
```suggestion
	    if (appDefinition.getSpec().getMonitor() != null && appDefinition.getSpec().getMonitor().getPort() > 0) {
```